cmd/thv-registry-api/app/serve.go

```diff
@@ -90,85 +88,97 @@ func getKubernetesConfig() (*rest.Config, error) {
 	return kubeConfig.ClientConfig()
 }
 
-// buildProviderConfig creates provider configuration based on command-line flags
-func buildProviderConfig() (*service.RegistryProviderConfig, error) {
-	configMapName := viper.GetString("from-configmap")
-	filePath := viper.GetString("from-file")
-	registryName := viper.GetString("registry-name")
+// createFileProviderForSyncedData creates a file provider that reads from the synced storage location
+// All source types (git, configmap, api, file) are synced to ./data/registry.json by the sync manager
+func createFileProviderForSyncedData(registryName string) (*service.RegistryProviderConfig, error) {
+	return &service.RegistryProviderConfig{
+		FilePath:     "./data/registry.json",
+		RegistryName: registryName,
+	}, nil
+}
 
-	// Validate mutual exclusivity
-	if configMapName != "" && filePath != "" {
-		return nil, fmt.Errorf("--from-configmap and --from-file flags are mutually exclusive")
+func runServe(_ *cobra.Command, _ []string) error {
+	ctx := context.Background()
+	address := viper.GetString("address")
+
+	// Initialize controller-runtime logger to suppress warnings
+	log.SetLogger(zap.New(zap.UseDevMode(false)))
+
+	logger.Infof("Starting registry API server on %s", address)
+
+	// Load and validate configuration (now required)
+	configPath := viper.GetString("config")
+	// TODO: Validate the path to avoid path traversal issues
+	cfg, err := config.NewConfigLoader().LoadConfig(configPath)
+	if err != nil {
+		return fmt.Errorf("failed to load configuration: %w", err)
 	}
+	if err := cfg.Validate(); err != nil {
+		return fmt.Errorf("invalid configuration: %w", err)
+	}
+	logger.Infof("Loaded configuration from %s (registry: %s, source: %s)",
+		configPath, cfg.GetRegistryName(), cfg.Source.Type)
 
-	// Require one of the flags
-	if configMapName == "" && filePath == "" {
-		return nil, fmt.Errorf("either --from-configmap or --from-file flag is required")
+	// Get Kubernetes config (needed for both sync manager and deployment provider)
+	k8sRestConfig, err := getKubernetesConfig()
```

Comment:
I don't think we should be creating a Kubernetes client here, we should only be doing it if the configmap is provided in the config. I'm happy to do this in my PR though.